src/runtime/pkg/oci/utils.go

```diff
@@ -926,14 +929,24 @@ func addHypervisorNetworkOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfi
 
 func addHypervisorInitdataOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfig) error {
 	if value, ok := ocispec.Annotations[vcAnnotations.Initdata]; ok {
-		initdataToml, err := base64.StdEncoding.DecodeString(value)
+		initdataTomlGzip, err := base64.StdEncoding.DecodeString(value)
 		if err != nil {
+			ociLog.Errorf("initdata uncompress failed! %v", err)
 			return err
 		}
 
-		initdataStr := string(initdataToml)
+		gzipReader, err := gzip.NewReader(bytes.NewReader(initdataTomlGzip))
+		if err != nil {
+			return fmt.Errorf("create gzip reader failed: %v", err)
+		}
+		defer gzipReader.Close()
 
-		sbConfig.HypervisorConfig.Initdata = initdataStr
+		initdataToml, err := io.ReadAll(gzipReader)
+		if err != nil {
+			return fmt.Errorf("uncompressing initdata with gzip error: %v", err)
+		}
+
+		initdataStr := string(initdataToml)
```

Comment:
I wonder if we should pass the initdata into the guest as a gzip as well, especially given the size limitations. Actually we may want to keep it as a gzip all the way to Trustee. We could do this in a future iteration, though.

Reply:
This is do-able. How about marking it as a TODO because the user interface will not change.